# Ok Cupid Dataset EDA

This dataset was picked on the name alone to do an initial clean and EDA.
The dataset is completly unfamiliar to me - I want to practice familiarising myself with the set and seeing what I can learn about it, to see what further investigation can stem from it. 

In [103]:
# initial imports and check file is readable
import pandas as pd
okcupid = pd.read_csv('OkCupid_kaggle_dataset.zip')


In [104]:
pd.set_option('display.max_columns', None) # use this to display ALL columns and not get a ... midway
print(okcupid.head((10)))

   age     status sex orientation       body_type               diet  \
0   22     single   m    straight  a little extra  strictly anything   
1   35     single   m    straight         average       mostly other   
2   38  available   m    straight            thin           anything   
3   23     single   m    straight            thin         vegetarian   
4   29     single   m    straight        athletic                NaN   
5   29     single   m    straight         average    mostly anything   
6   32     single   f    straight             fit  strictly anything   
7   31     single   f    straight         average    mostly anything   
8   24     single   f    straight             NaN  strictly anything   
9   37     single   m    straight        athletic    mostly anything   

       drinks      drugs                          education  \
0    socially      never      working on college/university   
1       often  sometimes              working on space camp   
2    socially     

In [105]:
okcupid.shape

(59946, 31)

This set has 59946 rows and 31 columns

## Duplicates check

In [106]:
duplicates = okcupid.duplicated()
print(duplicates.value_counts())

False    59946
dtype: int64


The number of false duplicates equals the number of rows - i.e. there are no dupicated rows

## Tidying/Wrangling

In [107]:
print(okcupid.columns)

Index(['age', 'status', 'sex', 'orientation', 'body_type', 'diet', 'drinks',
       'drugs', 'education', 'ethnicity', 'height', 'income', 'job',
       'last_online', 'location', 'offspring', 'pets', 'religion', 'sign',
       'smokes', 'speaks', 'essay0', 'essay1', 'essay2', 'essay3', 'essay4',
       'essay5', 'essay6', 'essay7', 'essay8', 'essay9'],
      dtype='object')


In [108]:
okcupid_variables = pd.DataFrame(okcupid.info())

# okcupid_variables = pd.DataFrame(okcupid.info()) #converted to a DataFrame to send to an Excel file


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59946 entries, 0 to 59945
Data columns (total 31 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   age          59946 non-null  int64  
 1   status       59946 non-null  object 
 2   sex          59946 non-null  object 
 3   orientation  59946 non-null  object 
 4   body_type    54650 non-null  object 
 5   diet         35551 non-null  object 
 6   drinks       56961 non-null  object 
 7   drugs        45866 non-null  object 
 8   education    53318 non-null  object 
 9   ethnicity    54266 non-null  object 
 10  height       59943 non-null  float64
 11  income       59946 non-null  int64  
 12  job          51748 non-null  object 
 13  last_online  59946 non-null  object 
 14  location     59946 non-null  object 
 15  offspring    24385 non-null  object 
 16  pets         40025 non-null  object 
 17  religion     39720 non-null  object 
 18  sign         48890 non-null  object 
 19  smok

As there are alot of variables to consider types and further actions for, I'd like to track these in an Excel sheet
To do this, I plan on exporting okcupid.info() to an Excel sheet directly


In [109]:
#from xlsxwriter import Workbook
#okcupid_variables.to_excel('okcupid_variables.xlsx', engine='xlsxwriter')
#okcupid_variables.to_csv('okcupid_variables.csv', sep=',', header=True) # works but loses all the data

In [110]:
#print(okcupid_variables.head()) # shows okcupid_variables is actually empty, so I didnt lose data in the CSV there just isnt any

This is something I want to keep in mind but will shelve for now as it'll take me more time to learn than save me at the moment.

Next I'll inspect each variable

0 'age' appears complete with Dtype int64 - ready to investiagte numerically
1 'status' - complete, object - check the unique types with .value_counts()
2 & 3 'sex' & 'orientation' - complete, object - check the unique types with .value_counts()

4 'body_type' - 54650 entries, therefore incomplete - check the unique types with .value_counts()
5 'diet' - 35551 - .value_counts()


In [111]:
okcupid.status.value_counts()

single            55697
seeing someone     2064
available          1865
married             310
unknown              10
Name: status, dtype: int64

'Status' is Relationship Status.
Categories are:
* Single
* Seeing somone
* Available
* Married
* Unknown - this one needs further investigation: unknown as in profile hasn't added their status? (from code snippet below I'd say yes)

In [112]:
#status_unknowns = okcupid[okcupid.status=='unknown']
#print(status_unknowns)
#print(len(status_unknowns))


In [113]:
okcupid.sex.value_counts()

m    35829
f    24117
Name: sex, dtype: int64

Sex is only in two vales: 'm' or 'f'

In [114]:
okcupid.orientation.value_counts()

straight    51606
gay          5573
bisexual     2767
Name: orientation, dtype: int64

Orientation is 'straight', 'gay' and 'bisexual'
Seen here large majority is straight, followed by gay then bisexual

In [115]:
okcupid.body_type.value_counts()

average           14652
fit               12711
athletic          11819
thin               4711
curvy              3924
a little extra     2629
skinny             1777
full figured       1009
overweight          444
jacked              421
used up             355
rather not say      198
Name: body_type, dtype: int64

average           14652, 
fit               12711, 
athletic          11819, 
thin               4711, 
curvy              3924, 
a little extra     2629, 
skinny             1777, 
full figured       1009, 
overweight          444, 
jacked              421, 
used up             355, 
rather not say      198

In [116]:
okcupid.diet.value_counts()

mostly anything        16585
anything                6183
strictly anything       5113
mostly vegetarian       3444
mostly other            1007
strictly vegetarian      875
vegetarian               667
strictly other           452
mostly vegan             338
other                    331
strictly vegan           228
vegan                    136
mostly kosher             86
mostly halal              48
strictly halal            18
strictly kosher           18
halal                     11
kosher                    11
Name: diet, dtype: int64

Variety of diet types, expaned by some having 'mostly...' and 'strictly...' categories

In [117]:
okcupid.drinks.value_counts()

socially       41780
rarely          5957
often           5164
not at all      3267
very often       471
desperately      322
Name: drinks, dtype: int64

In [118]:
okcupid.drugs.value_counts()

never        37724
sometimes     7732
often          410
Name: drugs, dtype: int64

In [119]:
okcupid.education.value_counts()

graduated from college/university    23959
graduated from masters program        8961
working on college/university         5712
working on masters program            1683
graduated from two-year college       1531
graduated from high school            1428
graduated from ph.d program           1272
graduated from law school             1122
working on two-year college           1074
dropped out of college/university      995
working on ph.d program                983
college/university                     801
graduated from space camp              657
dropped out of space camp              523
graduated from med school              446
working on space camp                  445
working on law school                  269
two-year college                       222
working on med school                  212
dropped out of two-year college        191
dropped out of masters program         140
masters program                        136
dropped out of ph.d program            127
dropped out

33 different categories, ranging from 23959 to 11 entries

In [120]:
okcupid.ethnicity.value_counts()

white                                                                 32831
asian                                                                  6134
hispanic / latin                                                       2823
black                                                                  2008
other                                                                  1706
                                                                      ...  
middle eastern, indian, white                                             1
asian, middle eastern, black, white, other                                1
asian, middle eastern, indian, hispanic / latin, white, other             1
black, native american, indian, pacific islander, hispanic / latin        1
asian, black, indian                                                      1
Name: ethnicity, Length: 217, dtype: int64

217 different values - many mixes of ethnicities 

In [121]:
okcupid.height.describe()

count    59943.000000
mean        68.295281
std          3.994803
min          1.000000
25%         66.000000
50%         68.000000
75%         71.000000
max         95.000000
Name: height, dtype: float64

In [122]:
okcupid.height.min()

1.0

Height looks to have some erroneous entries as the min value is 1. 

Assuming unit to be inches

In [138]:
profile_heights = okcupid[okcupid.height < 50]
print(profile_heights.height)
print(len(profile_heights))

7098     36.0
7786     43.0
12193     8.0
12330    36.0
18583    36.0
18832     3.0
20010    36.0
20334    42.0
22963    36.0
23767     6.0
26191    47.0
27485    49.0
29934    48.0
30498    36.0
31158    36.0
36801    37.0
37111     9.0
44218    37.0
45570    36.0
45959     1.0
47583    43.0
47793    36.0
48280    26.0
49022    48.0
55600    36.0
56287     4.0
57606    36.0
Name: height, dtype: float64
27


In [128]:
import seaborn as sns
import matplotlib.pyplot as plt

In [139]:
#sns.histplot(x='height', data=profile_heights)
#plt.show()

There are 4 entries under 10 inches. Investigate what to do with these results. Look at the rest of their entries

In [141]:
profile_heights = okcupid[okcupid.height < 10]
print(profile_heights)

       age  status sex orientation body_type               diet       drinks  \
12193   30  single   f    straight       NaN  mostly vegetarian     socially   
18832   39  single   m    straight  athletic                NaN     socially   
23767   32  single   f    straight      thin    mostly anything  desperately   
37111   25  single   m    straight       fit    mostly anything     socially   
45959   36  single   m    straight       NaN                NaN   very often   
56287   25  single   f    straight       NaN                NaN        often   

       drugs                          education  \
12193    NaN          graduated from space camp   
18832  never     graduated from masters program   
23767    NaN            working on ph.d program   
37111  never  graduated from college/university   
45959  never  graduated from college/university   
56287  often              working on space camp   

                                ethnicity  height   income  \
12193              

Other that several haveing no income (not significant when considered with below) the rest of the entries are not suspiscious

Income looks to be broken down into income brackets with -1 being the entry for no income given (and also having the highest count)

In [142]:
okcupid.income.value_counts()

-1          48442
 20000       2952
 100000      1621
 80000       1111
 30000       1048
 40000       1005
 50000        975
 60000        736
 70000        707
 150000       631
 1000000      521
 250000       149
 500000        48
Name: income, dtype: int64

In [158]:
okcupid.job.value_counts()
print('num of entries: ')
print(okcupid.job.value_counts().sum())

null_jobs = okcupid[okcupid.job.isnull()]
print(null_jobs)


num of entries: 
51748
       age     status sex orientation body_type               diet    drinks  \
2       38  available   m    straight      thin           anything  socially   
6       32     single   f    straight       fit  strictly anything  socially   
8       24     single   f    straight       NaN  strictly anything  socially   
10      35  available   m    straight   average    mostly anything  socially   
28      30     single   m    straight       fit                NaN  socially   
...    ...        ...  ..         ...       ...                ...       ...   
59926   28     single   f    straight  athletic    mostly anything  socially   
59933   30     single   f    straight       NaN                NaN  socially   
59935   33     single   f    straight     curvy           anything  socially   
59938   48     single   m    straight   average                NaN       NaN   
59940   31     single   f    straight       NaN                NaN  socially   

       drugs    

In [ ]:
last_online

In [ ]:
location 

In [ ]:
offspring 

In [ ]:
pets

In [ ]:
 religion 

In [ ]:
sign 

In [ ]:
smokes 

In [ ]:
speaks 

In [ ]:
essay0-9

In [ ]:
#okcupid.dtypes # done in .info()


In [ ]:
#okcupid.describe()